packages/admin-panel/src/pages/ProfilePage.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Tupaia MediTrak
+ * Copyright (c) 2017 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import { useForm } from 'react-hook-form';
+import { Button, TextField } from '@tupaia/ui-components';
+import { usePortalWithCallback } from '../utilities';
+import { Header } from '../widgets';
+import { updateProfile, getUser } from '../authentication';
+
+const Container = styled.section`
+  padding-top: 1rem;
+  padding-bottom: 1rem;
+  max-width: 460px;
+  margin: 3rem auto;
+`;
+
+const StyledButton = styled(Button)`
+  margin-top: 1rem;
+  padding-top: 0.8rem;
+  padding-bottom: 0.8rem;
+`;
+
+const ErrorMessage = styled.p`
+  color: ${props => props.theme.palette.error.main};
+`;
+
+const SuccessMessage = styled.p`
+  color: ${props => props.theme.palette.success.main};
+`;
+
+const ProfilePageComponent = React.memo(({ user, onUpdateProfile, getHeaderEl }) => {
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [successMessage, setSuccessMessage] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const { handleSubmit, register, errors } = useForm();
+  const HeaderPortal = usePortalWithCallback(<Header title={user.name} />, getHeaderEl);
+
+  const onSubmit = handleSubmit(async ({ firstName, lastName, role, employer }) => {
+    setIsLoading(true);
+    setErrorMessage(null);
+    try {
+      await onUpdateProfile(user.id, {
+        first_name: firstName,
+        last_name: lastName,
+        position: role,
+        employer: employer,
+      });
+      setIsLoading(false);
+      setSuccessMessage('Profile successfully updated.');
+    } catch (error) {
+      setErrorMessage(error.message);
+      setIsLoading(false);
```

Comment:
This is super nitpicky, but just for consistency...

```suggestion
      setIsLoading(false);
      setSuccessMessage('Profile successfully updated.');
    } catch (error) {
      setIsLoading(false);
      setErrorMessage(error.message);
```